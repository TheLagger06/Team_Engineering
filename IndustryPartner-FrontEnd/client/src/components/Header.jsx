import React from "react";
import logo3 from './logo3.png'
import 'bootstrap/dist/css/bootstrap.min.css';
import '../App.css';
import  Main from './TalentSpotlight/Main.jsx'
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { useCookies } from "react-cookie";
import axios from "axios";
import { ToastContainer, toast } from "react-toastify";

function Header() {
     const [cookies, setCookie, removeCookie] = useCookies([]);
    const logOut = () => {
    removeCookie("jwt");
    };
    
    return (
        <>
            <nav className="navbar navbar-expand-sm navbar-dark flex-sm-nowrap flex-wrap">
                <div className="container-fluid">
                    <a className="navbar-brand" href="/">
                        <img src={logo3}
                            alt="Digital futures logo" className="logoimage" /> DFX
                    </a>
                    <div className="navbar-collapse collapse flex-grow-1 justify-content-end" id="navbarNavAltMarkup">
                        <div className="navbar-nav">
                            {/* <a class="nav-item nav-link 1" href="#"> Talk to us {/*<span class="sr-only">(current)</span></a>*/}
                            {/* <a class="nav-item nav-link 2" href="#"> Home button </a> */}
                            {/* <a class="nav-item nav-link 3" href="#"> Your requirements </a> */}
                            <a className="nav-item nav-link 4" href="https://digitalfutures.com/contact/" > Talk to us</a>
                            <a className="nav-item nav-link 4" href="/"> Talent spotlight </a>
                            <a onClick={logOut} className="nav-item nav-link 4" href="/">Log out </a>

                        </div>
                    </div>
                </div>
            </nav>

        </>
    )
}


export default Header;



