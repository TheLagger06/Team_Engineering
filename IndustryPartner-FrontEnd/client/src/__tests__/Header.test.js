import { render, screen } from '@testing-library/react';

import Header from '../components/Header';


describe('Header component tests', () => {
    test(`Should match the snapshot`, () => {

        expect(render(<Header />)).toMatchSnapshot();


    })

});
