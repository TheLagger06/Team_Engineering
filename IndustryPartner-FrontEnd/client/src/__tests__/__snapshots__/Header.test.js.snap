// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Header-Comp"
      />
      <nav
        class="navbar navbar-expand-sm navbar-dark flex-sm-nowrap flex-wrap"
        style="width: 100%;"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="Digital futures logo"
              class="logoimage"
              src="logo3.png"
            />
             DFX
          </a>
          <div
            class="navbar-collapse collapse flex-grow-1 justify-content-end"
            id="navbarNavAltMarkup"
          >
            <div
              class="navbar-nav"
            >
              <a
                class="nav-item nav-link 4"
                href="https://digitalfutures.com/contact/"
              >
                 Talk to us
              </a>
              <a
                class="nav-item nav-link 4"
                href="/"
              >
                 Talent spotlight 
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <div
      class="Header-Comp"
    />
    <nav
      class="navbar navbar-expand-sm navbar-dark flex-sm-nowrap flex-wrap"
      style="width: 100%;"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Digital futures logo"
            class="logoimage"
            src="logo3.png"
          />
           DFX
        </a>
        <div
          class="navbar-collapse collapse flex-grow-1 justify-content-end"
          id="navbarNavAltMarkup"
        >
          <div
            class="navbar-nav"
          >
            <a
              class="nav-item nav-link 4"
              href="https://digitalfutures.com/contact/"
            >
               Talk to us
            </a>
            <a
              class="nav-item nav-link 4"
              href="/"
            >
               Talent spotlight 
            </a>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
